relay-metrics/src/statsd.rs

```diff
@@ -153,16 +162,17 @@ impl GaugeMetric for MetricGauges {
 ///
 /// In order to keep this low-cardinality, we only enumerate a handful of well-known, high volume
 /// names. The rest gets mapped to "other".
-pub fn metric_name_tag(value: &str) -> &str {
-    if [
+pub(crate) fn metric_name_tag(value: &str) -> &'static str {
+    if let Some(value) = [
         "c:sessions/session@none",
         "s:sessions/user@none",
         "s:sessions/error@none",
         "d:transactions/duration@millisecond",
         "s:transactions/user@none",
         "c:transactions/count_per_root_project@none",
     ]
-    .contains(&value)
+    .into_iter()
+    .find(|x| x == &value)
```

Comment:
This was refactored in https://github.com/getsentry/relay/pull/2041, but using `find` instead of `contains` allows us to return a static str and prevents allocations on the caller side.

Reply:
would unwrap_or("other") work too ?

Reply:
Would you assign that to `value`? Because then you cannot compare with other values anymore (see comparisions below).